framework/src/modules/interoperability/mainchain/commands/cc_update.ts

```diff
@@ -12,18 +12,433 @@
+	public addDependencies(args: CrossChainCommandDependencies) {
+		this._bftAPI = args.bftAPI;
+		this._validatorsAPI = args.validatorsAPI;
+	}
+	public async verify(
+		context: CommandVerifyContext<CrossChainUpdateTransactionParams>,
+	): Promise<VerificationResult> {
+		const { params: txParams, transaction } = context;
+		const errors = validator.validate(crossChainUpdateTransactionParams, context.params);
+
+		if (errors.length > 0) {
+			return {
+				status: VerifyStatus.FAIL,
+				error: new LiskValidationError(errors),
+			};
+		}
+
+		const partnerChainIDBuffer = getIDAsKeyForStore(txParams.sendingChainID);
+		const partnerChainStore = context.getStore(transaction.moduleID, STORE_PREFIX_CHAIN_DATA);
+		const partnerChainAccount = await partnerChainStore.getWithSchema<ChainAccount>(
+			partnerChainIDBuffer,
+			chainAccountSchema,
+		);
+
+		// Liveness of Partner Chain
+		if (partnerChainAccount.status === CHAIN_TERMINATED) {
+			return {
+				status: VerifyStatus.FAIL,
+				error: new Error(`Sending partner chain ${txParams.sendingChainID} is terminated.`),
+			};
+		}
+		const interoperabilityStore = this.getInteroperabilityStore(context.getStore as StoreCallback);
+		const isChainLive = await interoperabilityStore.isLive(partnerChainIDBuffer, Date.now());
+		if (partnerChainAccount.status === CHAIN_ACTIVE && !isChainLive) {
+			return {
+				status: VerifyStatus.FAIL,
+				error: new Error(`Sending partner chain ${txParams.sendingChainID} is not live.`),
+			};
+		}
+
+		// Liveness Requirement for the First CCU
+		if (partnerChainAccount.status === CHAIN_REGISTERED) {
+			// Certificate must not be empty for first CCU
+			if (!txParams.certificate) {
+				return {
+					status: VerifyStatus.FAIL,
+					error: new Error(
+						`Sending partner chain ${txParams.sendingChainID} is in registered status so certificate cannot be empty.`,
+					),
+				};
+			}
+		}
+		const decodedCertificate = codec.decode<Certificate>(certificateSchema, txParams.certificate);
+		const invalidCertificateErrors = validator.validate(certificateSchema, decodedCertificate);
+		if (invalidCertificateErrors.length > 0) {
+			return {
+				status: VerifyStatus.FAIL,
+				error: new Error('Certificate does not follow valid certificate schema.'),
+			};
+		}
+		if (
+			txParams.activeValidatorsUpdate &&
+			txParams.activeValidatorsUpdate.length !== 0 &&
+			txParams.newCertificateThreshold > BigInt(0)
+		) {
+			// Non-empty certificate
+			if (!txParams.certificate) {
+				return {
+					status: VerifyStatus.FAIL,
+					error: new Error('Certificate cannot be empty.'),
+				};
+			}
+
+			// params.activeValidatorsUpdate has the correct format
+			for (let i = 0; i < txParams.activeValidatorsUpdate.length; i += 1) {
+				const currentValidator = txParams.activeValidatorsUpdate[i];
+				const nextValidator = txParams.activeValidatorsUpdate[i + 1];
+				if (currentValidator.blsKey.byteLength !== VALID_BLS_KEY_LENGTH) {
```

Comment:
Not related to this PR, but we should align on the usage of `byteLength` vs `length`. (I think `byteLength` is correct, but `length` is used everywhere else)